### PR TITLE
Fixed User Agent Parsing

### LINKED ISSUE #35: Fix Agent Details
Different Eth2 clients returning agent details in different format. Due to this, information stored in the database is inconsistent. In some case `agent name` is empty. Possibly because some clients return agent details in `/name/version/os` format.

In addition to that,
-  we need to store representable the OS names. i.e. convert `aarch64-macos` to `Mac` and `x86_64-linux` to `Linux`
- we need to store just the client version. i.e. convert `v1.0.2-f7183098` to `v1.0.2`

### PR DESCRIPTION
**Pull request by sadiq1971**
Fixed #35

### CODE REVIEW

File: models/peer.go [hunk 1]
```diff
@@ -20,11 +20,36 @@ import (
...
+
+const (
+	OSLinux   OS = "linux"
+	OSMAC     OS = "mac"
+	OSWindows OS = "windows"
+	OSUnknown OS = "unknown"
+)
+
 // UserAgent holds peer's client related info
 type UserAgent struct {
```
Comment (dhyaniarun1993) [inline models/peer.go:49]: I think we should save the raw agent data as well for debugging purpose. What do think?

Comment (sadiq1971) [reply models/peer.go:49]: If we need it for debugging we can print it in log instead of storing in db.

Comment (dhyaniarun1993) [reply models/peer.go:49]: Okay let's do that for now. Although, It could be painful to go through the logs.

Comment (sadiq1971) [reply models/peer.go:49]: Yeah, It is painful. But there might not be any issue as  I tested almost all possible agent details.

**Review by dhyaniarun1993 — APPROVED** [review]

**Pull request merged**